### PR TITLE
Expose column auto size API feature request

### LINKED ISSUE #2025: igx-grid - cannot auto-size columns by double-clicking in IE11
### Description  
Describe the issue.

 * igniteui-angular version: 6.1.1
 * browser: IE11

### Steps to reproduce  

1. Go to https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid_column_resizing.html
2. Double click the right side of the "Compan..." header. 

### Result  
The column width is a little bit expanded. But it is not enough.
image

### Expected result  
This is a result of Chrome. The "Company Name" column is fully expanded.
image

### PR DESCRIPTION
**Pull request by SAndreeva**
Closes #2025  .  

1. Expose `autosize()` method on `IgxColumnComponent`.
2. Fix #2025.
3. Expose two helper methods (in utils) for measuring string length: using Range and using Canvas. Current implementation of auto sizing uses the one with Range.

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grid/column-resizing.spec.ts [hunk 1]
```diff
@@ -492,55 +474,51 @@ describe('IgxGrid - Deferred Column Resizing', () => {
...
-    //     tick();
-    //     fixture.detectChanges();
+        expect(grid.columns[1].width).toEqual('150px');
 
-    //     expect(grid.columns[1].width).toEqual("207px");
-    //     expect(fixture.componentInstance.count).toEqual(2);
-    //     expect(fixture.componentInstance.column).toBe(grid.columns[1]);
-    //     expect(fixture.componentInstance.prevWidth).toEqual("150");
-    //     expect(fixture.componentInstance.newWidth).toEqual("207px");
+        headers[1].componentInstance.resizeArea.nativeElement.dispatchEvent(dblclick);
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grid/column-resizing.spec.ts]: Please check if `dblclick` event could be triggered through `UIInteractions.simulateMouseEvent`.

File: projects/igniteui-angular/src/lib/grid/grid-header.component.ts [hunk 2]
```diff
@@ -145,6 +151,7 @@ export class IgxGridHeaderComponent implements OnInit, DoCheck, AfterViewInit {
             this.zone.runOutsideAngular(() => {
                 this.resizeArea.nativeElement.addEventListener('mouseover', this.onResizeAreaMouseOver.bind(this));
                 this.resizeArea.nativeElement.addEventListener('mousedown', this.onResizeAreaMouseDown.bind(this));
+                this.resizeArea.nativeElement.addEventListener('dblclick', this.onResizeAreaDblClick.bind(this));
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grid/grid-header.component.ts]: These three methods (event handlers) could be made private.

File: projects/igniteui-angular/src/lib/grid/column.component.ts [hunk 3]
```diff
@@ -934,6 +935,38 @@ export class IgxColumnComponent implements AfterContentInit {
...
+    /**
+     * Returns a reference to the header of the column.
+     * ```typescript
+     * let column = this.grid.columnList.filter(c => c.field === 'ID')[0];
+     * let headerCell = column.headerCell;
+     * ```
+     * @memberof IgxColumnComponent
+     */
+    get headerCell(): IgxGridHeaderComponent {
+        return flatten(this.grid.headerList.toArray()).filter((h) => h.column.index === this.index)[0];
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grid/column.component.ts]: Please check if the array has length > 0.

File: projects/igniteui-angular/src/lib/grid/grid.component.ts [hunk 4]
```diff
@@ -4122,4 +4129,58 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
...
+     * Returns the size (in pixels) of the longest currently visible cell, including the header cell.
+     * ```typescript
+     * @ViewChild('grid') grid: IgxGridComponent;
+     *
+     * let column = this.grid.columnList.filter(c => c.field === 'ID')[0];
+     * let size = this.grid.getgetLongestCell(column);
+     * ```
+     * @memberof IgxColumnComponent
+     */
+    public getLargestCellWidth(column: IgxColumnComponent): string {
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grid/grid.component.ts]: You could move this method to `IgxColumnComponent`.

File: projects/igniteui-angular/src/lib/grid/grid.component.ts [hunk 5]
```diff
@@ -4122,4 +4129,58 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
...
     }
 
+    /**
+     * @hidden
+     * Returns the size (in pixels) of the longest currently visible cell, including the header cell.
+     * ```typescript
+     * @ViewChild('grid') grid: IgxGridComponent;
+     *
+     * let column = this.grid.columnList.filter(c => c.field === 'ID')[0];
+     * let size = this.grid.getgetLongestCell(column);
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grid/grid.component.ts]: `getgetLongestCell` looks strange to me. Is it a typo?

**Comment by npaunov** [comment]
The check if the browser is IE does not always work.

**Comment by npaunov** [comment]
#2025 still reproduces when the grid's and columns' widths are in percentage.
This is related to #1951.

**Pull request merged**